presto-expressions/src/main/java/com/facebook/presto/expressions/LogicalRowExpressions.java

```diff
@@ -370,6 +379,86 @@ public RowExpression filterConjuncts(RowExpression expression, Predicate<RowExpr
         return combineConjuncts(conjuncts);
     }
 
+    public static GroupingElement removeGroupingElementPrefix(GroupingElement groupingElement, Optional<Identifier> removablePrefix)
+    {
+        if (groupingElement instanceof SimpleGroupBy) {
+            boolean prefixRemoved = false;
+            ImmutableList.Builder<Expression> rewrittenColumns = ImmutableList.builder();
+            for (Expression column : groupingElement.getExpressions()) {
+                Expression processedColumn = removeExpressionPrefix(column, removablePrefix);
+                if (processedColumn != column) {
+                    prefixRemoved = true;
+                }
+                rewrittenColumns.add(processedColumn);
+            }
+            if (prefixRemoved) {
+                return new SimpleGroupBy(rewrittenColumns.build());
+            }
+        }
+        return groupingElement;
+    }
+
+    public static SortItem removeSortItemPrefix(SortItem sortItem, Optional<Identifier> removablePrefix)
```

Comment:
nit: consider returning early when removablePrefix is not present. There is no point to visit all the children when it is not present.